code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -318,16 +318,26 @@
 	icon_state = "hardsuit0-rd"
 	item_color = "rd"
 	unacidable = 1
+	var/onboard_hud_enabled = 0 //stops conflicts with another diag HUD
 	max_heat_protection_temperature = FIRE_SUIT_MAX_TEMP_PROTECT
 	armor = list(melee = 10, bullet = 5, laser = 10, energy = 5, bomb = 100, bio = 100, rad = 60)
 
-/obj/item/clothing/head/helmet/space/hardsuit/rd/equipped(mob/user, slot)
-	user.scanner.Grant(user)
+/obj/item/clothing/head/helmet/space/hardsuit/rd/equipped(mob/living/carbon/human/user, slot)
 	..(user, slot)
+	user.scanner.Grant(user)
+	if(user.glasses && istype(user.glasses, /obj/item/clothing/glasses/hud/diagnostic))
+		user << ("<span class='warning'>Your [user.glasses] prevents you using [src]'s diagnostic visor HUD.</span>")
+	else
+		onboard_hud_enabled = 1
+		var/datum/atom_hud/DHUD = huds[DATA_HUD_DIAGNOSTIC]
+		DHUD.add_hud_to(user)
 
-/obj/item/clothing/head/helmet/space/hardsuit/rd/dropped(mob/user)
-	user.scanner.devices -= 1
+/obj/item/clothing/head/helmet/space/hardsuit/rd/dropped(mob/living/carbon/human/user)
 	..(user)
+	user.scanner.devices -= 1
+	if(onboard_hud_enabled && !(user.glasses && istype(user.glasses, /obj/item/clothing/glasses/hud/diagnostic)))
```

Comment:
this is a hacky fix, and it doesn't even solve the same issue when the items are switched (removing your diag glasses while wearing the helmet).